File: ExponeaSDKMauiIOS/ExponeaSDKMauiIOS/MethodResult.swift
//
//  MethodResult.swift
//  ExponeaSDKMauiIOS
//
//  Created by Adam Mihalik on 14/08/2023.
//

import Foundation

@objc(MethodResult)
public class MethodResult : NSObject {
    @objc
    public let success: Bool
    @objc
    public let data: String
    @objc
    public let error: String
    
    init(success: Bool, data: String, error: String) {
        self.success = success
        self.data = data
        self.error = error
    }
    
    public static func success(_ data: String?) -> MethodResult {
        return MethodResult(success: true, data: data ?? "", error: "")
    }
    
    public static func failure(_ message: String) -> MethodResult {
        return MethodResult(success: false, data: "", error: message)
    }
    
    public static func unsupportedMethod(_ method: String) -> MethodResult {
        return .failure("Method \(method) is currently unsupported")
    }
}

File: BloomreachSDKMauiAndroid/lib/src/main/java/com/bloomreach/sdk/maui/android/MethodResult.kt
package com.bloomreach.sdk.maui.android

data class MethodResult(
    val success: Boolean,
    val data: String,
    val error: String
) {
    companion object {
        fun success(data: String? = null): MethodResult = MethodResult(
            success = true,
            data = data ?: "",
            error = ""
        )
        fun failure(message: String): MethodResult = MethodResult(
            success = false,
            data = "",
            error = message
        )
    }
}
